### DISCUSSION TITLE
[General] Overall EA31337 philosophy

### DISCUSSION THREAD

**Discussion by cgava** [EA31337/EA31337#405] [Q&A]
Seen from away, my understanding of EA31337 classes is to provide a generic wrapper that enables to use the component created (Indicator, Exper Advisors, Strategies) usable in different context:

- Executed inside MT4
- Executed inside MT5
- Executed outside Metatrader with historic values

Am I right ?

**Comment by kenorb** [comment]
### Compatibilty

The EA31337 project (EA+libraries) aims to run on both MT4 and MT5 platforms with maximal compatibility when possible (sometimes it's not, e.g. lack of support for databases, matrices/matrix, OpenCL, async order calls and other in MT4).

On the other hand it's clear than MQ wants to push towards MT5, by dropping MT4 support at some point (e.g. they've removed public endpoints, removed MQL4 compilation from MTE5, etc.). So as long there is an interest, MT4 should be be supported to some extend.

Initially support for both was very challenging under one codebase and many developers didn't believe it can be fully done due to lot of technical challenges. At the end the project was built against the odds to show others the way how this can be done.

Now, the project is taking it into another level by adding slowly the support for C++ and WebAssembly for platform agnostic integrations (EA31337-classes & FX31337-wasm projects). This will open possibilities of trading capabilities outside of the platform (such as using Colab, CLI, Node.JS, Python and many more).

### Framework

As part of the overall project, EA31337-classes repository (licensed under GPL-3) is a framework/library to allow other programmers to develop trading solutions in the more convenient way. It allows to code trading solutions for real markets making trading no longer a taboo topic. A lot of companies or people develop their own propriety systems which takes a lot of effort and usually their system don't work at the end and a lot of code lands into a bin. So this project is to help people (by already solving a lot of challenges), so they don't have to code from scratch. Since the project is licensed under GPL-3, we expect others to contribute back upon their challenges (by disclosing modified source when you distribute the software), so we are all working collaboratively at the common goals.
